apps/server/src/modules/user/repo/role.repo.ts

```diff
@@ -0,0 +1,110 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { Role, IPermissionsAndRoles } from '../entity';
+
+interface CacheEntry {
+	[name: string]: { value: Role; validUntil: number };
+}
+
+// TODO: no query handled at the moment
+// TODO: write test and move
+// TODO: replace with generall solution if it is added
+export class RoleCache {
+	cache: CacheEntry;
+
+	name: string;
+
+	// we do not registred changes by calling the old endpoint
+	// also the clear interval should short
+	clearInterval: 60000;
+
+	constructor(name = '') {
+		this.cache = {};
+		this.name = name;
+	}
+
+	get(selector: string): Role | undefined {
+		const entry = this.cache[selector];
+		if (!entry || entry.validUntil < Date.now()) return undefined;
```

Comment:
I would have expected the cache to fetch the entry if it doesnt have it already, to have the cache control the caching strategy.
I do see the point of not letting the cache know how to access the data, but im sure there is solutions for that.